Dashboard/Connectors/Serial/SerialPortWrapper.cs

```diff
@@ -65,4 +65,21 @@ public bool IsConnected
             return timeSinceLastMessage.TotalSeconds < ConnectionTimeout && _serialPort.IsOpen;
         }
     }
+
+    public void Write(string? data)
+    {
+        if (_serialPort.IsOpen)
+        {
+            _serialPort.WriteTimeout = 50;
+            try
+            {
+                _serialPort.Write(data);
```

Comment:
q: does this block the main thread? do we need to consider opening a thread that writes to the port also?

Reply:
Now that you've brought it up, I think it can block the main thread, I'll open a thread to handle this.